semgrep-core/Matching/Generic_vs_generic.ml

```diff
@@ -910,6 +910,28 @@ and m_literal a b =
     -> fail ()
 (*e: function [[Generic_vs_generic.m_literal]] *)
 
+and m_wrap_m_int_opt (a1, a2) (b1, b2) =
+  match a1, b1 with
+  (* iso: semantic equivalence of value! 0x8 can match 8 *)
+  | Some i1, Some i2 when i1 =|= i2 -> return ()
+  (* if the integers (or floats) were too large or were using
+   * a syntax OCaml int_of_string could not parse,
+   * we default to a string comparison *)
+  | _ ->
+      let a1 = Parse_info.str_of_info a2 in
```

Comment:
The integer literal may come from constant propagation and in that case the token was created with `Parse_info.fake_info`. Passing one of these fake tokens to `str_of_info` will make Semgrep crash with a `NoTokenLocation` exception.

There are different things we could do here... for example, we could change

```ocaml
| Some i1, Some i2 when i1 =|= i2 -> return ()
```

to

```ocaml
| Some i1, Some i2 -> if i1 =|= i2 then return () else fail ()
```

We could perhaps also make `str_of_info` get the first string argument of `FakeTokStr`? And perhaps `token_location_of_info` could return throw a warning and return a dummy `token_location`?